CorsixTH/Lua/dialogs/resizables/hotkey_assign.lua

```diff
@@ -764,328 +630,67 @@ function UIHotkeyAssign:close()
-  current_pos_y = current_pos_y + 1
-  -- ingame_recallPosition_8
-  self:addBevelPanel(panel_x_pos, self.panel_pos_table_y[current_pos_y], panel_width, panel_height, col_shadow, col_bg, col_bg) : setLabel(_S.hotkey_window.ingame_recallPosition_8)
-  self.hotkey_buttons["ingame_recallPosition_8"] = self:addBevelPanel(panel_x_pos+panel_width, self.panel_pos_table_y[current_pos_y], panel_width, panel_height, col_hotkeybox, col_highlight, col_shadow)
-    :makeHotkeyBox(function() self:confirm_func("ingame_recallPosition_8") end, nil):setText( string.upper(array_join(ui.app.hotkeys["ingame_recallPosition_8"], "+")) )
-  current_pos_y = current_pos_y + 1
-  -- ingame_recallPosition_9
-  self:addBevelPanel(panel_x_pos, self.panel_pos_table_y[current_pos_y], panel_width, panel_height, col_shadow, col_bg, col_bg) : setLabel(_S.hotkey_window.ingame_recallPosition_9)
-  self.hotkey_buttons["ingame_recallPosition_9"] = self:addBevelPanel(panel_x_pos+panel_width, self.panel_pos_table_y[current_pos_y], panel_width, panel_height, col_hotkeybox, col_highlight, col_shadow)
-    :makeHotkeyBox(function() self:confirm_func("ingame_recallPosition_9") end, nil):setText( string.upper(array_join(ui.app.hotkeys["ingame_recallPosition_9"], "+")) )
-  current_pos_y = current_pos_y + 1
-  -- ingame_recallPosition_0
-  self:addBevelPanel(panel_x_pos, self.panel_pos_table_y[current_pos_y], panel_width, panel_height, col_shadow, col_bg, col_bg) : setLabel(_S.hotkey_window.ingame_recallPosition_0)
-  self.hotkey_buttons["ingame_recallPosition_0"] = self:addBevelPanel(panel_x_pos+panel_width, self.panel_pos_table_y[current_pos_y], panel_width, panel_height, col_hotkeybox, col_highlight, col_shadow)
-    :makeHotkeyBox(function() self:confirm_func("ingame_recallPosition_0") end, nil):setText( string.upper(array_join(ui.app.hotkeys["ingame_recallPosition_0"], "+")) )
-end
-
-function UIHotkeyAssign_storeRecallPos:close()
-  UIResizable.close(self)
-  if self.mode == "menu"  then
-    self.ui:addWindow(UIHotkeyAssign(self.ui, "menu"))
+  local current_pos_x = 1
+  for _, section in ipairs(key_options.sections) do
+    local current_pos_y = 1
+    self:addBevelPanel(self.panel_pos_table_x[current_pos_x], self.panel_pos_table_y[current_pos_y], panel_width*2, panel_height, col_caption):setLabel(section.title)
+    current_pos_y = current_pos_y + 1
+    for _, key in ipairs(section.keys) do
+      self:addBevelPanel(self.panel_pos_table_x[current_pos_x], self.panel_pos_table_y[current_pos_y], panel_width, panel_height, col_shadow, col_bg, col_bg) : setLabel(_S.hotkey_window[key])
+    self.hotkey_buttons[key] = self:addBevelPanel(self.panel_pos_table_x[current_pos_x]+panel_width, self.panel_pos_table_y[current_pos_y], panel_width, panel_height, col_hotkeybox, col_highlight, col_shadow)
+      :makeHotkeyBox(function() self:confirm_func(key) end, nil):setText( string.upper(array_join(app_hotkeys[key], "+")) )
```

Comment:
Wrong indentation, also 2nd line should have double indentation.

Also, if you add a local x and y for
``` lua
local x = self.panel_pos_table_x[current_pos_x]
local y = self.panel_pos_table_y[current_pos_y]
```
that would go a long way to get lines with a more readable length, and give space around the expression operators in it

Reply:
Done